### PR TITLE
Adding back ability to crop on fov.get_image()

### PR DESCRIPTION
**Pull request by shanaxel42**
Unblocks @ambrosejcarr on seqFish work for now but will eventually be overridden by #1328

### CODE REVIEW

**Comment by shanaxel42** [comment]
> Needs tests. :)

Was the test in test_experiment.py not enough?

**Comment by ttung** [comment]
> Was the test in test_experiment.py not enough?

I swear it wasn't there when I reviewed it, apologies. :(

File: starfish/core/imagestack/parser/crop.py [hunk 1]
```diff
@@ -58,6 +58,31 @@ def _add_permitted_axes(self, axis_type: Axes, permitted_axis: int) -> None:
         if axis_type == Axes.ZPLANE and self._permitted_zplanes:
             self._permitted_zplanes.add(permitted_axis)
 
+    def further_crop(self, further_crop_params) -> "CropParameters":
+        if further_crop_params._permitted_chs and \
```
Comment (ttung) [inline starfish/core/imagestack/parser/crop.py]: when you have a multiline condition clause, i think it's ok to do

```
if (xxxx
    and yyyy ...
):
```

File: starfish/core/imagestack/parser/crop.py [hunk 2]
```diff
@@ -58,6 +58,31 @@ def _add_permitted_axes(self, axis_type: Axes, permitted_axis: int) -> None:
         if axis_type == Axes.ZPLANE and self._permitted_zplanes:
             self._permitted_zplanes.add(permitted_axis)
 
+    def further_crop(self, further_crop_params) -> "CropParameters":
```
Comment (ttung) [inline starfish/core/imagestack/parser/crop.py]:
```suggestion
    def further_crop(self, further_crop_params: "CropParameters") -> "CropParameters":
```

A simple docblock here would be good.

File: starfish/core/imagestack/parser/crop.py [hunk 3]
```diff
@@ -58,6 +58,31 @@ def _add_permitted_axes(self, axis_type: Axes, permitted_axis: int) -> None:
...
+                not further_crop_params._permitted_rounds.issubset(self._permitted_rounds):
+            raise ValueError("Can only provide further round crop params within the existing set of"
+                             "round crop params")
+        else:
+            self._permitted_rounds = further_crop_params._permitted_rounds
+
+        if further_crop_params._permitted_zplanes and \
+                not further_crop_params._permitted_zplanes.issubset(self._permitted_zplanes):
+            raise ValueError("Can only provide further zplane crop params within the existing set "
+                             "of z_plane crop params")
```
Comment (ttung) [inline starfish/core/imagestack/parser/crop.py]:
```suggestion
                             "of zplane crop params")
```

^^ to match your earlier reference.

File: starfish/core/imagestack/parser/crop.py (hunk 1 above)
Comment (ttung) [inline starfish/core/imagestack/parser/crop.py]: You could rewrite the three if-else as one single if-else inside a loop if you want.  It's debatable if it's better.

**Comment by dganguli** [comment]
Sorry for chiming in a bit late here, but I'd like to revisit the name "Crop" in this new context. In the field of image processing / image analysis, cropping implies selecting a subset of an image in *spatial coordinates*, e.g., X, Y, and Z. If I understand correctly, this PR _also_ allows me to select a subset of an image in non-spatial coordinates, e.g., R, C. 

I definitely want this functionality, but I find the name CropParameters and all the references to crop to be confusing. Here are some other ideas:

 ```crop``` ->  ```select``` or ```subset``` or ```index```

correspondingly, 
```CropParameters``` -> ```SelectionParameters``` or ```SubsetParameters``` or ```AxesParameters```

**Comment by shanaxel42** [comment]
> Sorry for chiming in a bit late here, but I'd like to revisit the name "Crop" in this new context. In the field of image processing / image analysis, cropping implies selecting a subset of an image in _spatial coordinates_, e.g., X, Y, and Z. If I understand correctly, this PR _also_ allows me to select a subset of an image in non-spatial coordinates, e.g., R, C.
> 
> I definitely want this functionality, but I find the name CropParameters and all the references to crop to be confusing. Here are some other ideas:
> 
> `crop` -> `select` or `subset` or `index`
> 
> correspondingly,
> `CropParameters` -> `SelectionParameters` or `SubsetParameters` or `AxesParameters`

@dganguli does the suggestion that @ambrosejcarr made about just passing in the r/ch/z parameters directly work for your concerns?

**Comment by dganguli** [comment]
@shanaxel42 @ambrosejcarr whoops! Yeah it seems like Ambrose and I are saying the same thing. After re-reading, I agree with Ambrose. Sorry!

**Comment by dganguli** [comment]
i do think, we should avoid using the term ```crop``` if it refers to more than manipulating the X,Y,Z axes though. ambrose's solution indeed avoids this.

**Comment by shanaxel42** [comment]
merging this as an intermediate step but immediately working on  #1328 and will address @ttung testing case

**Pull request merged**